znai/src/main/java/com/twosigma/znai/website/markups/SphinxParsingConfiguration.java

```diff
@@ -48,6 +49,16 @@ public Path fullPath(ComponentsRegistry componentsRegistry, Path root, TocItem t
         return root.resolve(tocItem.getDirName()).resolve(tocItem.getFileNameWithoutExtension() + "." + filesExtension());
     }
 
+    @Override
```

Comment:
Duplicated code.
Move up to MarkupParsingConfiguration and make it default method or make MarkupParsingConfiguration an abstract class.

Reply:
@byc1234 making it default or abstract would work but won't be ideal as the implementation outside znai codebase will most likely override it (mine implementation does a different implementation).  I tried to extract to re-usable utils to make the code small. It does look like code duplicate, but the way I look at it is, two implementation just happen to delegate to have same logic and delegate to the same function.  Sphinx configuration and Markdown configuration don't have anything in common on semantic level so I would prefer to avoid binding them with some common implementation parent. 

Wonder if we should add a new `toc.findTocItem(path)` signature to automatically extract dir-name and file-name from the given path. This way implementations will be even smaller and probably will look more like call delegation than code duplication. 
On the other hand overloaded `toc.findTocItem` may get confused with the `tocItemByPath`.

Reply:
Make sense for the response of my comment.

I don't have any opinion about a signature of findTocItem. I am fine with either the current one or your suggested one.